pkg/auth/webapp/session_authflow.go

```diff
@@ -774,6 +774,19 @@ func (s *AuthflowScreenWithFlowResponse) Advance(p string, result *Result) {
 	result.RedirectURI = u.String()
 }
 
+// RedirectToFinish is a fix for https://linear.app/authgear/issue/DEV-1793/investigate-sign-in-directly-with-httpsaccountsportalauthgearcom-crash
+// We need Turbo to visit /finish with a full browser redirect,
+// so CSP and connect-src will not kick in.
+func (s *AuthflowScreenWithFlowResponse) RedirectToFinish(route string, result *Result) {
+	q := url.Values{}
+	q.Set(AuthflowQueryKey, s.Screen.StateToken.XStep)
+	u, _ := url.Parse(route)
+	u.RawQuery = q.Encode()
```

Comment:
And as this part shares the code with `Advance` and they are expected to be same, I think we better make a function to do it.

Reply:
OK. Let me change that.